src/python/pants/bsp/util_rules/targets.py

```diff
@@ -67,6 +77,8 @@ async def bsp_workspace_build_targets(
     for response in responses:
         result.extend(response.targets)
     result.sort(key=lambda btgt: btgt.id.uri)
+    output_digest = await Get(Digest, MergeDigests([r.digest for r in responses]))
+    workspace.write_digest(output_digest, path_prefix=".pants.d/bsp")
```

Comment:
Might be good to move the `.pants.d/bsp` path to a constant, or onto a context object... mostly to highlight the connection between the usages. These correspond to the URIs created in `src/python/pants/backend/scala/bsp/rules.py` ...?

Reply:
I'm going to move it into BSPGoal as a config option in a future PR. There's now enough places in the BSP rules to extract it out into one place.